include/zio/zio_defs.h

```diff
@@ -0,0 +1,396 @@
+	 *
+	 * Expressed in meters (m).
+	 */
+	DEV_CHANNEL_POS_DIST_SURF,
+
+	/**
+	 * Proximity to an external plane or object.
+	 *
+	 * Expressed in millimeters (mm).
+	 */
+	DEV_CHANNEL_POS_PROX,
+
+	/**
+	 * Encapsulates a raw NMEA sentence from a GPS-type device
+	 *
+	 * Expressed as a null-terminated string, with the exact string
+	 * contents determined by NMEA standard. NMEA sentences are generally
+	 * limited to 79 characters plus the null-termination character.
+	 */
+	DEV_CHAN_POS_NMEA,
+
+	/**
+	 * An Euler angle, which consists of the three familiar X/Y/Z
+	 * headings.
+	 *
+	 * Expressed as a 3-vector (XYZ) of degrees.
+	 */
+	DEV_CHAN_ORIENT_EULER,
+
+	/**
+	 * A quaternion, which is generally used to represent a value in
+	 * three dimensional space, avoiding the problem of gimbal lock
+	 * associated with 3-vector Euler angles.
+	 *
+	 * Quaternions consist of four elements: a 3-vector and a scalar.
+	 */
+	DEV_CHAN_ORIENT_QUAT,
+
+	/**
+	 * An indication of tilt in degrees on a single axis relative to an
+	 * unspecified reference plane.
+	 *
+	 * Expressed in degrees (+/-180).
+	 */
+	DEV_CHAN_ORIENT_TILT,
+
+
+	/**
+	 * Pressure, which is a measurement of the force applied
+	 * perpendicularly to a specific surface area.
+	 *
+	 * Expressed in hectopascal (hPa)
+	 */
+	DEV_CHANNEL_PRESS,
+
+	/**
+	 * Barometric (or 'atmospheric') pressure, which is a measurement of
+	 * the pressure within the atmosphere. Related to the concept of
+	 * 'hydrostatic pressure' caused by the weight of the air above the
+	 * point of measure.
+	 *
+	 * TODO: Would millibar (mb) be a better unit choice, or should this
+	 * remain consist with the more general pressure channel unit (hPa)?
+	 *
+	 * Expressed in hectopascal (hPa)
+	 */
+	DEV_CHANNEL_PRESS_BAROM,
+
+	/**
+	 * Particulate matter.
+	 *
+	 * Expressed as micrograms per cubic meter (ug/m^3).
+	 */
+	DEV_CHAN_PART_MAT,
+
+	/**
+	 * Generic parts per million measurement. The exact type of matter
+	 * being measured should be indicated with an appropriate channel
+	 * attribute.
+	 *
+	 * Expressed as parts per million (ppm).
+	 */
+	DEV_CHAN_PPM,
+
+	/**
+	 * Detected CO2 levels.
+	 *
+	 * Expressed as parts per million (ppm).
+	 */
+	DEV_CHAN_PPM_CO2,
+
+	/**
+	 * Detected volatile organic compound (voc) levels.
+	 *
+	 * Expressed as parts per million (ppm).
+	 */
+	DEV_CHAN_PPM_VOC,
+
+        /**
+	 * User-defined channel. This can be used at the application level for
+         * whetever purpose the driver requires, provided it does not overlap
+         * with one of the existing standard channel definitions.
+         *
+         * If multiple user defined channels are present, they should be
+         * clearly distinguished by use of appropriate channel attributes.
+	 *
+	 * Expressed as a user-defined unit and scale. The exact unit type and
+         * scale should be clearly documented in the driver making use of this
+         * channel type.
+	 */
+	DEV_CHAN_USER_DEF,
```

Comment:
I think this should be set to some fixed value, leaving plenty of values below and above it for extending predefined channel types and allowing user defined expansion. Something like DEV_CHAN_USER_DEF = 32768
if we're going for a 16 bit channel type

Reply:
Yes, I agree ... I was going to put 0xFF since I think 8 bits is likely enough channels in the real world, but was hesitating just wondering if we should have some other variant like `DEV_CHAN_???_DEF` that is logically the final value.  Also ... maybe there are strong feelings to make this 16-bits, even though I think we're doing something wrong if we need to define more than 200 entries in the enum here! I'm OK with setting this to `0xFF`, though, which was my first thought as well.

Reply:
Provisionally resolved with https://github.com/zephyrproject-rtos/zephyr/pull/14245/commits/d57cd9b21f41c2f33e4e4e2a1f861606a19b6938 ... can be changed to 16-bit value if deemed appropriate.